community/rust/alpine-target.patch

```diff
@@ -23,229 +23,99 @@ and distribution-specific quirks instead of polluting the main musl target of
-+    base.max_atomic_width = Some(64);
-+    base.stack_probes = true;
-+    base.crt_static_default = false;
-+    base.static_position_independent_executables = true;
-+    base.need_rpath = true;
-+
-+    Ok(Target {
-+        llvm_target: "x86_64-alpine-linux-musl".to_string(),
-+        target_endian: "little".to_string(),
-+        target_pointer_width: "64".to_string(),
-+        target_c_int_width: "32".to_string(),
-+        data_layout: "e-m:e-i64:64-f80:128-n8:16:32:64-S128".to_string(),
-+        arch: "x86_64".to_string(),
-+        target_os: "linux".to_string(),
-+        target_env: "musl".to_string(),
-+        target_vendor: "alpine".to_string(),
-+        linker_flavor: LinkerFlavor::Gcc,
-+        options: base,
-+    })
++    let mut base = super::arm_unknown_linux_musleabihf::target()?;
++
++    base.llvm_target = "armv6-alpine-linux-musleabihf".to_string();
++    base.target_vendor = "alpine".to_string();
++    base.options.crt_static_default = false;
++    base.options.static_position_independent_executables = true;
++    base.options.need_rpath = true;
++
++    Ok(base)
 +}
 --- /dev/null
-+++ b/src/librustc_target/spec/aarch64_alpine_linux_musl.rs
-@@ -0,0 +1,36 @@
-+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
-+// file at the top-level directory of this distribution and at
-+// http://rust-lang.org/COPYRIGHT.
-+//
-+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
-+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
-+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
-+// option. This file may not be copied, modified, or distributed
-+// except according to those terms.
-+
-+use crate::spec::{LinkerFlavor, Target, TargetOptions, TargetResult};
++++ b/src/librustc_target/spec/armv7_alpine_linux_musleabihf.rs
+@@ -0,0 +1,13 @@
++use crate::spec::TargetResult;
 +
 +pub fn target() -> TargetResult {
-+    let mut base = super::linux_musl_base::opts();
-+    base.max_atomic_width = Some(128);
-+    base.crt_static_default = false;
-+    base.static_position_independent_executables = true;
-+    base.need_rpath = true;
-+
-+    Ok(Target {
-+        llvm_target: "aarch64-alpine-linux-musl".to_string(),
-+        target_endian: "little".to_string(),
-+        target_pointer_width: "64".to_string(),
-+        target_c_int_width: "32".to_string(),
-+        data_layout: "e-m:e-i8:8:32-i16:16:32-i64:64-i128:128-n32:64-S128".to_string(),
-+        arch: "aarch64".to_string(),
-+        target_os: "linux".to_string(),
-+        target_env: "musl".to_string(),
-+        target_vendor: "alpine".to_string(),
-+        linker_flavor: LinkerFlavor::Gcc,
-+        options: TargetOptions {
-+            abi_blacklist: super::arm_base::abi_blacklist(),
-+            .. base
-+        },
-+    })
++    let mut base = super::armv7_unknown_linux_musleabihf::target()?;
++
++    base.llvm_target = "armv7-alpine-linux-musleabihf".to_string();
```

Comment:
The linker errors suggest the float-abi needs adjusting.

Reply:
Oh, but do we build another armv7 target into LLVM? It was my impression that that's our only armv7 target